### ISSUE TITLE
Use legacy WLW plugins

### ISSUE THREAD

**Issue by giggio** [OpenLiveWriter/OpenLiveWriter#454]
Is there any guidance on how to use legacy plugins? Even if it means manually moving files? Will they work at all? And if not, how to upgrade them?

**Comment by lextm** [comment]
Duplicate to #19

**Comment by giggio** [comment]
@lextm Not really, #19 is about a plugins gallery, this one is about running WLW plugins on OLW. These are complementary, but completely different things.

**Comment by ScottIsAFool** [comment]
@giggio They won't work at all unfortunately. They are dependent on a file that has now changed name and signature. I have a collection of 30+ plugins already converted to work with OLW, but as yet, we have no good way of distributing or installing them. But this is something we are keen to get out soon.

**Comment by ronmartblog** [comment]
@ScottIsAFool - Is there any way these plug ins could be stored in GitHub or some readme file that says where to get them and user contributions on how to install them? Something like a wiki perhaps?

I really miss not having my WLW Templates when blogging.

**Comment by ScottIsAFool** [comment]
@ronmartblog If you want to give a project I was working on a try, then:

1. Go to https://github.com/ScottIsAFool/LiveWriterPluginManager/releases/tag/v0.2.0.1 and run setup.exe, this will install the Live Writer Plugin Manager. 
2. You'll then need to download a plugin package, which in your case, can be got from https://1drv.ms/u/s!Am2IZiFT4KKztOZLcsIehl618Dxj7Q.
3. Once that is installed, run it as administrator.  
4. In the Plugin Manager, click on the Browse for plugin button and find where you saved the olwpkg file. 

The plugin manager should then install the plugin for you and it will appear when you run Live Writer.

**Comment by rkingett** [comment]
I would love to try this manager but when I tried it it would not work for me because my screen reader, NVDA, was not even reading me the title bar. :(

**Comment by bc3tech** [comment]
@ScottIsAFool got Syntax Highlighter handy? trying to get this installed [via the instructions on his site](https://richhewlett.com/2015/12/22/openlivewriter-version-of-syntax-highlighter-plugin-available-now/) and it's not working w/ OLW 0.6

**Comment by ScottIsAFool** [comment]
@bc3tech His instructions work for me (using option 2), with an added step he may or may not know about. You need to right click the dll and go to the properties, then unblock the file before OLW will read it.

**Comment by bc3tech** [comment]
@ScottIsAFool hm. double-checked the Unblock, added to reg, still no dice.  thoughts?

**Comment by ScottIsAFool** [comment]
Can you show me a screenshot of regedit where you've added it into the registry.

**Comment by bc3tech** [comment]
got it. either OLW took more time to load it than i was allotting before passing judgement, or my "unblock" of the accompanying .txt file (not just the .dll) was the kicker.

**Comment by bc3tech** [comment]
@ScottIsAFool
Could you point me to a "how-to" for converting old WLW plugins to OLW-compatible ones? I want to take the source for [_Code Snippet_](http://wlwplugincollection.codeplex.com/releases/view/9392) and get it over. I like it better than Syntax Highlighter.

**Comment by ScottIsAFool** [comment]
I've already converted most plugins, if you go to https://github.com/ScottIsAFool/LiveWriterPluginManager/releases/tag/v0.2.0.2 and install the plugin manager, then download the package from https://1drv.ms/u/s!Am2IZiFT4KKztoRbUHBLA7zrJKa0Ow you can install it via that.

**Comment by bc3tech** [comment]
@ScottIsAFool 
thoughts on this?
![image](http://content.screencast.com/users/bc3tech/folders/Snagit/media/e3b68d5f-1846-45ac-a98d-a1293eebfd6d/09.23.2016-15.02.png)

**Comment by ScottIsAFool** [comment]
Hmmm. Check the dlls and make sure they're not unblocked. I didn't think they would be, but I guess they could be.

**Comment by bc3tech** [comment]
nope, not blocked.

**Comment by ScottIsAFool** [comment]
I didn't think they should be. I'm seeing the same thing, let me look into this (although will be next week now)

**Comment by bc3tech** [comment]
@ScottIsAFool 
np. do you have the folder of all the plugins you've converted publicly available or you just send out links to ones people want as they come up? ;)
Might be worth a Wiki post?

**Comment by ScottIsAFool** [comment]
There's a rough plan for plugins, I just need to finish something for that plugin manager before we can put them all live. At the moment, I'm making them available as and when people ask for them.

**Comment by jphellemons** [comment]
@ScottIsAFool I am a big fan of precode which is on codeplex, it's old but does the job. They have a WLW plugin too. Is there an easy fix? Or should I wait for the next release of Open Live Writer?
(which is epic by the way, great job all!) ps. I am currently using OLW from the Microsoft/Windows Store.

**Comment by ScottIsAFool** [comment]
@jphellemons at the moment there's no way for the store version to have plugins (we have an idea on how to do this though).

**Comment by jphellemons** [comment]
Great! Thanks @ScottIsAFool 👍

**Comment by simonsabin** [comment]
@ScottIsAFool I've tried to follow this for the code snippet but I'm getting OpenLiveWriter,1.2124,None,00005,13-Feb-2017 17:21:10.602,"System.IO.FileLoadException: Could not load file or assembly 'OpenLiveWriter.Api, Version=0.5.1.4, Culture=neutral, PublicKeyToken=null' or one of its dependencies. A strongly-named assembly is required. (Exception from HRESULT: 0x80131044)
File name: 'OpenLiveWriter.Api, Version=0.5.1.4, Culture=neutral, PublicKeyToken=null'

Do you have an update plugin with the later api.

**Comment by shanselman** [comment]
Thanks for reporting this and for helping Open Live Writer over the years. We are doing an issue-bankruptcy cleanup so we can make the repository maintainable again and focus on current, actionable work. This older issue is being closed as part of that cleanup, not because it was unimportant. If this still reproduces in the current code or is still relevant to the revitalization work, please open a fresh issue with current details and we will triage it going forward.
